spec/02-integration/05-proxy/01-router_spec.lua

```diff
@@ -882,6 +882,119 @@ for _, strategy in helpers.each_strategy() do
       end)
     end)
 
+    describe("path prefixes and regexes", function()
+      local routes
+
+      lazy_setup(function()
+        routes = insert_routes {
+          {
+            strip_path = true,
+            paths      = { "/root/fix" },
+          },
+          {
+            strip_path = true,
+            paths      = { "/root/(fixture)" },
+          },
+          {
+            strip_path = true,
+            paths      = { "/root2/fixture" },
+          },
+          {
+            strip_path = true,
+            paths      = { "/root2/fix" },
+          },
+          {
+            strip_path = true,
+            paths      = { "/root2/(fixture)" },
+          }
+
+        }
+      end)
+
+      lazy_teardown(function()
+        remove_routes(routes)
+      end)
+
+      it("prioritizes regexes over prefixes", function()
+        local res = assert(proxy_client:send {
+          method  = "GET",
+          path    = "/root/fixture",
+          headers = {
+            ["kong-debug"] = 1,
+          }
+        })
+
+        assert.res_status(200, res)
+
+        assert.equal(routes[2].id,           res.headers["kong-route-id"])
+        assert.equal(routes[2].service.id,   res.headers["kong-service-id"])
+        assert.equal(routes[2].service.name, res.headers["kong-service-name"])
+      end)
```

Comment:
This test is a duplicate of [this unit test](https://github.com/Kong/kong/blob/master/spec/01-unit/08-router_spec.lua#L452). Btw, it is preferred that router tests be added to the unit test suite pointed above, instead here. This test suite, while it may have _some_ overlap with the above unit suite, relates more to integration of the router within Kong at runtime.